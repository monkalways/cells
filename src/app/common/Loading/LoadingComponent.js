import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress, withStyles } from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  size: PropTypes.number,
};

const defaultProps = {
  size: 100,
};

const LoadingComponent = ({ classes, size }) => (
  <div className={classes.container}>
    <CircularProgress className={classes.progress} size={size} />
  </div>
);

LoadingComponent.propTypes = propTypes;
LoadingComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  container: {
    display: 'flex',
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  progress: {
    margin: theme.spacing.unit * 2,
  },
}))(LoadingComponent);
