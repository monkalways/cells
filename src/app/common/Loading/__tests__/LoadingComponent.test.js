import React from 'react';
import { shallow } from 'enzyme';

import { CircularProgress } from '@material-ui/core';
import { LoadingComponent } from '../LoadingComponent';

describe('LoadingComponent', () => {
  let classes;
  let size;

  beforeEach(() => {
    classes = {
      container: 'container',
      progress: 'progress',
    };
    size = null;
  });

  it('should render with default size 100', () => {
    const wrapper = shallow(<LoadingComponent classes={classes} />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find(CircularProgress)).toHaveProp('size', 100);
  });

  it('should render with provided size', () => {
    size = 50;
    const wrapper = shallow(<LoadingComponent classes={classes} size={size} />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find(CircularProgress)).toHaveProp('size', size);
  });
});
