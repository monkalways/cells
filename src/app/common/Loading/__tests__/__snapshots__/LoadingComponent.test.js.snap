// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingComponent should render with default size 100 1`] = `
<div
  className="container"
>
  <WithStyles(CircularProgress)
    className="progress"
    size={100}
  />
</div>
`;

exports[`LoadingComponent should render with provided size 1`] = `
<div
  className="container"
>
  <WithStyles(CircularProgress)
    className="progress"
    size={50}
  />
</div>
`;
